core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -182,7 +244,14 @@ internal suspend fun processTransientSettingEvents(
                 }
             }
         }
+        // todo(kc): Enable torch toggle while recording in progress
+        if (currentCameraState.value.videoRecordingState is VideoRecordingState.Active &&
+            newTransientSettings.flashMode == FlashMode.ON
```

Comment:
You'll want to adapt the code from #283 to keep track of when the torch is on locally. It ensures the torch is toggled properly when the device supports it.